packages/test/test-end-to-end-tests/src/test/container.spec.ts

```diff
@@ -518,4 +519,76 @@ describeNoCompat("Container", (getTestObjectProvider) => {
             "container is not connected after rapid disconnect() + connect()",
         );
     });
+
+    it("Disposing container does not send deltaManager readonly event", async () => {
+        const container = await createConnectedContainer();
+
+        let run = 0;
+        container.deltaManager.on("readonly", () => run++);
+
+        container.dispose();
+        assert.strictEqual(run, 0, "DeltaManager should not send readonly event on container dispose");
+    });
+
+    it("Closing container sends deltaManager readonly event", async () => {
+        const container = await createConnectedContainer();
+
+        let run = 0;
+        container.deltaManager.on("readonly", () => run++);
+
+        container.close();
+        assert.strictEqual(run, 1, "DeltaManager should send readonly event on container close");
+    });
+
+    it("Closing container should not send dispose events", async () => {
+        const container = await createConnectedContainer();
+        const dataObject = await requestFluidObject<ITestDataObject>(container, "default");
+
+        let deltaManagerDisposed = 0;
+        let deltaManagerClosed = 0;
+        let runtimeDispose = 0;
+        (container.deltaManager as any).on("disposed", () => deltaManagerDisposed++);
+        (container.deltaManager as any).on("closed", () => deltaManagerClosed++);
+        (dataObject._context.containerRuntime as ContainerRuntime).on("dispose", () => runtimeDispose++);
+
+        container.close();
+        assert.strictEqual(deltaManagerDisposed, 0, "DeltaManager should not send disposed event on container close");
+        assert.strictEqual(deltaManagerClosed, 1, "DeltaManager should send closed event on container close");
+        assert.strictEqual(runtimeDispose, 0, "ContainerRuntime should not send dispose event on container close");
+    });
+
+    it("Disposing container should send dispose events", async () => {
+        const container = await createConnectedContainer();
+        const dataObject = await requestFluidObject<ITestDataObject>(container, "default");
+
+        let deltaManagerDisposed = 0;
+        let deltaManagerClosed = 0;
+        let runtimeDispose = 0;
+        (container.deltaManager as any).on("disposed", () => deltaManagerDisposed++);
+        (container.deltaManager as any).on("closed", () => deltaManagerClosed++);
+        (dataObject._context.containerRuntime as ContainerRuntime).on("dispose", () => runtimeDispose++);
+
+        container.dispose();
+        assert.strictEqual(deltaManagerDisposed, 1, "DeltaManager should send disposed event on container dispose");
+        assert.strictEqual(deltaManagerClosed, 0, "DeltaManager should not send closed event on container dispose");
+        assert.strictEqual(runtimeDispose, 1, "ContainerRuntime should send dispose event on container dispose");
+    });
+
+    it("Closing then disposing container should send close and dispose events", async () => {
```

Comment:
What about disposing then closing?  Is that valid?

Reply:
Disposing acts as a superset to closing, so disposing then closing is not valid